documentation/docs/mysql.md

```diff
@@ -0,0 +1,1090 @@
+# MySQL Enterprise 8.4 + Cosmian KMS Integration
+
+## Executive Summary
+
+This document provides a comprehensive guide for integrating MySQL Enterprise with Cosmian Key Management System (KMS) using the `keyring_okv` plugin for Transparent Data Encryption (TDE). This lab demonstrates secure key management and data encryption at rest in a controlled environment.
+
+---
+
+## Table of Contents
+
+1. [Version Requirements](#version-requirements)
+2. [Technical Specifications](#technical-specifications)
+3. [Architecture Overview](#architecture-overview)
+4. [Prerequisites](#prerequisites)
+5. [Installation and Configuration](#installation-and-configuration)
+6. [Capabilities and Features](#capabilities-and-features)
+7. [Testing and Validation](#testing-and-validation)
+8. [Advanced Resilience Testing](#advanced-resilience-testing)
+9. [Troubleshooting](#troubleshooting)
+10. [References](#references)
+
+---
+
+## Version Requirements
+
+### Minimum Versions for TDE with keyring_okv
+
+| Component | Minimum Version | Notes |
+|-----------|-----------------|-------|
+| **MySQL Enterprise** | 8.0+ | keyring_okv plugin available in Enterprise Edition |
+| **Cosmian KMS** | 5.14+ | Full KMIP 1.1 compatibility |
+| **KMIP Protocol** | 1.1 | Standard protocol version supported by keyring_okv |
+| **OpenSSL** | 3.0+ | For TLS 1.3 and mTLS certificate support |
+
+**Note:** This lab uses **MySQL 8.4.7** and **Cosmian KMS 5.14+** which are the stable, production-ready versions recommended for new deployments.
+
+---
+
+## Technical Specifications
+
+### System Requirements
+
+| Component | Version | Specification |
+|-----------|---------|---------------|
+| **MySQL Enterprise Server** | 8.4.7-commercial | Generic Linux x86_64 binary (glibc 2.28) |
+| **Cosmian KMS** | 5.14+ | KMIP 1.1 server with socket support |
+| **Operating System** | Ubuntu 24.04 LTS (Noble) | x86_64 architecture |
```

Comment:
The KMS server binary `cosmian_kms` is compatible on all Debian-based distros from Debian 10, so I do not know what is this OS requirement